spec/system/distribution_system_spec.rb

```diff
@@ -206,6 +257,46 @@
       end.to change { Distribution.count }.by(-1)
     end
 
+    context "when delivery method is not shipped" do
+      it "should not display shipping_cost field" do
+        click_on "Edit", match: :first
+
+        # if element not found it will throw exception
+        expect { page.find_by_id("shipping_cost_div", wait: 2) }.to raise_error(Capybara::ElementNotFound)
+      end
+    end
+
+    context "when delivery method is shipped" do
+      let(:delivery_method) { "shipped" }
+
+      context "when shipping cost is negative" do
+        it "should not update distribution, display shipping_cost field and show error message" do
+          click_on "Edit", match: :first
+
+          # to check if shipping_cost field exist
+          expect(page.find_by_id("shipping_cost_div")).not_to be_nil
+
+          fill_in "Shipping cost", with: -12.05
+          click_on "Save", match: :first
+          expect(page.find(".alert-danger")).to have_content "Sorry, we weren't able to save the distribution. Shipping cost cannot be negative!"
+        end
+      end
+
+      context "when shipping cost is positive" do
+        it "should update distribution and display shipping_cost field" do
+          click_on "Edit", match: :first
+
+          # to check if shipping_cost field exist
+          expect(page.find_by_id("shipping_cost_div")).not_to be_nil
+
+          fill_in "Shipping cost", with: 12.05
+          click_on "Save", match: :first
+          expect(page).to have_content "Distributions"
+          expect(page.find(".alert-info")).to have_content "Distribution updated!"
+        end
+      end
+    end
```

Comment:
See above.

Reply:
removed test for negative shipping_cost scnerio